### PR TITLE
feat: add config and choose provider screens

### PR DESCRIPTION
**Pull request by peelar**
https://github.com/saleor/apps/assets/44495184/83e24057-94e0-4bff-b688-2d6a10b05b23

## Next up:
- [ ] - add/edit provider forms
- [ ] - display providers in a table (I started it in this PR)

### CODE REVIEW

**Comment by lkostrowski** [comment]
Video - great! <3

File: apps/taxes/src/lib/theme-synchronizer.test.tsx [hunk 1]
```diff
@@ -38,7 +38,8 @@ vi.mock("@saleor/macaw-ui", () => {
 });
 
 describe("ThemeSynchronizer", () => {
```
Comment (lkostrowski) [inline apps/taxes/src/lib/theme-synchronizer.test.tsx:40]: impl of this changed with new macaw, you can delete test and copy-paste impl from other app. eventually this must be extracted to shared

File: apps/taxes/src/modules/avatax/avatax-config.ts [hunk 2]
```diff
@@ -31,13 +53,18 @@ export const avataxInstanceConfigSchema = z.object({
...
 export type AvataxInstanceConfig = z.infer<typeof avataxInstanceConfigSchema>;
 
-export const obfuscateAvataxConfig = (config: AvataxConfig) => ({
+export const obfuscateAvataxConfig = (config: AvataxConfig): AvataxConfig => ({
   ...config,
-  username: obfuscateSecret(config.username),
-  password: obfuscateSecret(config.password),
+  credentials: {
+    ...config.credentials,
+    username: obfuscateSecret(config.credentials.username),
```
Comment (lkostrowski) [inline apps/taxes/src/modules/avatax/avatax-config.ts:60]: username should be obfuscated?

File: apps/taxes/src/modules/ui/app-breadcrumbs.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+  return breadcrumbs;
+};
+
+export const AppBreadcrumbs = () => {
+  const breadcrumbs = useBreadcrumbs();
+
+  return (
+    <Breadcrumbs>
+      {breadcrumbs?.map((breadcrumb) => (
```
Comment (lkostrowski) [inline apps/taxes/src/modules/ui/app-breadcrumbs.tsx:56]: If breadcrumbs dont exist, it means app is broken (your made paths wrong). If app is broken, it should not fallback, but throw. Thanks to that you can quickly figure out problem, especially that routes are *not* strictly typed. If you fallback, you will render blank / empty component and bug will be harder to find

File: apps/taxes/src/pages/providers/index.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+import Image from "next/image";
+import { useRouter } from "next/router";
+import { AvataxIcon, StripeTaxIcon, TaxJarIcon } from "../../assets";
+import { AppCard } from "../../modules/ui/app-card";
+import { AppColumns } from "../../modules/ui/app-columns";
+
+const Header = () => {
+  return (
+    <Box>
+      <Text __maxWidth={"360px"} __fontWeight={"400"} variant="body">
```
Comment (lkostrowski) [inline apps/taxes/src/pages/providers/index.tsx:11]: fontWeight should use what Macaw provides (no `__`, use allowed values)

**Review by lkostrowski — APPROVED** [review]

File: apps/taxes/src/pages/providers/index.tsx (hunk 4 above)
Comment (peelar) [reply apps/taxes/src/pages/providers/index.tsx:11]: Unfortunately, I have some issues with Macaw font weights. I will describe them to Macaw team.

File: apps/taxes/src/lib/theme-synchronizer.test.tsx (hunk 1 above)
Comment (peelar) [reply apps/taxes/src/lib/theme-synchronizer.test.tsx:40]: It's copied from invoices. I removed the test

File: apps/taxes/src/modules/avatax/avatax-config.ts (hunk 2 above)
Comment (lkostrowski) [reply apps/taxes/src/modules/avatax/avatax-config.ts:60]: But are you sure this is a secret? Its strange, I have never seen to obfuscate any name/nickname type of field

**Pull request merged**